.github/workflows/run-script-and-commit.yml

```diff
@@ -0,0 +1,45 @@
+name: Run Script and Commit Changes
+
+on:
+  workflow_call:
+    inputs:
+      script_call:
+        required: true
+        type: string
+      file_to_commit:
+        required: true
+        type: string
+      commit_message:
+        required: true
+        type: string
+      user_email:
+        required: true
+        type: string
+      user_name:
+        required: true
+        type: string
+
+jobs:
+  run_script_and_commit:
```

Comment:
extracted this as it will also contain logic for commit signing and configuring github PAT.